data/tutorials/language/1ms_02_dune.md

```diff
@@ -0,0 +1,241 @@
+let stratus = "strato" ^ latin_root ^ "us (Sc)"
+```
+
+**`lib/stratus.mli`**
+```ocaml
+val v : string
+val cumulus : string
+```
+**`lib/stratus.ml`**
+```ocaml
+let latin_root = "strat"
+let v = latin_root ^ "us (St)"
+let cumulus = latin_root ^ "ocumulus (Sc)"
+```
+
+The executable and the corresponding `dune` file need to be updated to use the defined library as a dependency.
+
+**`dune`**
+```lisp
+(executable
+  (name cloud)
+  (public_name nube)
+  (libraries wmo))
+```
+
+
+
+**Observations**:
+* Dune creates a module `Wmo` from the contents of folder `lib`.
+* The folder's name (here `lib`) is irrelevant.
+* The library name appears uncapitalised (`wmo`) in `dune` files:
+  - In its definition, in `lib/dune`
+  - When used as a dependency in `dune`
+
+## Library Wrapper Modules
+<!-- CONCEPT: Submodule -->
+
+By default, modules bundled into a library by Dune are wrapped into a module. It is possible to bypass Dune's behaviour by manually writing the wrapper file.
+
+This `lib/wmo.ml` file corresponds to the module that Dune automatically generated in the previous section.
+
+**`lib/wmo.ml`**
+```ocaml
+module Cumulus = Cumulus
+module Stratus = Stratus
+```
+
+Here is how this makes sense of these module definitions:
+- On the left-hand side, `module Cumulus` means module `Wmo` contains a submodule named `Cumulus`.
+- On the right-hand side, `Cumulus` refers to the module defined in the files `lib/cumulus.ml` and `lib/cumulus.mli`.
+
+Check with `dune exec nube` to ensure the program's behaviour is the same as in the previous section.
+
+When a library folder contains a wrapper module (here `wmo.ml`), it is the only one exposed. A file-based module that does not appear in the wrapper module is private.
+
+Using a wrapper file makes several things possible:
+- Have different public and internal names, `module CumulusCloud = Cumulus`
+- Define values in the wrapper module, `let ... = `
+- Expose module resulting from functor application, `module StringSet = Set.Make(String)`
+- Apply the same interface type to several modules without duplicating files
+
+## Include Subdirectories
+
+By default, Dune builds libraries from modules found in folders, but it doesn't look into subfolders. It is possible to change this behaviour.
+
+In this example, we create subdirectories and move files there.
+```shell
+$ mkdir lib/cumulus lib/stratus
+$ mv lib/cumulus.ml lib/cumulus/m.ml
+$ mv lib/cumulus.mli lib/cumulus/m.mli
+$ mv lib/stratus.ml lib/stratus/m.ml
+$ mv lib/stratus.mli lib/stratus/m.mli
+```
+
+Change from the default behaviour with the `include_subdirs` stanza.
+
+**`lib/dune`**
+```lisp
+(include_subdirs qualified)
+(library (name wmo))
+```
+
+Update the library wrapper to expose the modules created from the subdirectories.
+
+**`wmo.ml`**
+```ocaml
+module Cumulus = Cumulus.M
+module Stratus = Stratus.M
+```
+
+Check with `dune exec nube` that the behaviour of the program is the same as in the previous sections.
+
+## Conclusion
```

Comment:
Do you want to remove this section or complete it?

Reply:
It needs to be completed